src/lib/y2firewall/clients/auto.rb

```diff
@@ -166,6 +170,57 @@ def modified?
 
     private
 
+      # Import method for the first stage.
+      #
+      # Except a few edge cases complete firewall configuration should be done in first
+      # stage. Exceptions are installation over network (ssh / vnc) with second stage
```

Comment:
I would do always the import and configuration during the first stage but opening the ports as we do during the installation only if the second_stage is enabled.

At the end of the second stage we should close opened ports that were not declared in the profile, and this should be probably done checking some mark as a temporal file.

Reply:
Well I didn't want to fiddle with logic for adding / removing rules for ssh / vnc so I simply postpone enabling firewall into second stage if needed. Worked fine and logic is simple. It makes the system open in some cases during the second stage but it should be "short" period and in fact behaves the same like before in all cases.